apollo-api/src/main/java/com/spotify/apollo/RequestValue.java

```diff
@@ -88,7 +88,9 @@ public Request withHeader(String name, String value) {
   @Override
   public Request withHeaders(Map<String, String> additionalHeaders) {
     Map<String, String> headers = new LinkedHashMap<>(headers());
-    headers.putAll(additionalHeaders);
+    for (Map.Entry<String, String> header : additionalHeaders.entrySet()) {
+      headers.put(header.getKey().toLowerCase(), header.getValue());
+    }
```

Comment:
To @udoprog's point though, when looking up the header all that would really be required is to do a `s1.equalsIgnoreCase('s2')` and you can keep storing it exactly in the format that came in, which could be useful for other purposes.